source/iot/MqttClient.cpp

```diff
@@ -352,7 +403,10 @@ namespace Aws
                 return nullptr;
             }
 
-            if (!(*newConnection) || !newConnection->SetLogin("?SDK=CPPv2&Version=" AWS_CRT_CPP_VERSION, nullptr))
+            char username[100] = "";
+            sprintf(username, "%s%s%s%s", "?SDK=", config.m_sdkName.c_str(), "&Version=", config.m_sdkVersion.c_str());
```

Comment:
The string length of username is probably something that should be looked up with MQTT spec maximum. Maybe less than spec if the spec is unreasonably high.

Use snprintf() instead of sprintf().

Reply:
Made changes as requested.